File: koa/src/client/scripts/models/Recipe.js
/**
 * Represents cookbook recipe.
 * @constructor
 */
define(['lib/helpers'], function (helpers) {

    function Recipe() {
        this.id = helpers.uid();
        this.name = '';
        this.description = '';
        this.photoId = this.id;
        this.popularity = 0; // [1, 2, 3, ..., 10]
        this.complexity = 0; // [1, 2, 3, ..., 10]
        this.ingredients = []; // string[]
        this.steps = []; // string[]
    }

    return Recipe;
});

File: angular2/cookbook/Models/Recipe.cs
using System;
using System.Collections.Generic;

namespace cookbook.Models
{
    public class Recipe
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int Complexity { get; set; }
        public int Popularity { get; set; }

        public List<Step> Steps { get; set; }
        public List<Ingredient> Ingredients { get; set; }
    }
}